app/controllers/pieces_controller.rb

```diff
@@ -19,7 +19,7 @@ def update
   private
   
     def set_piece
-      @piece = Piece.find(params[:id])
+      @piece ||= Piece.find(params[:id])
```

Comment:
Controller attributes are fresh on every request, they have no life between requests (and if they did, you'd have other problems, as the `id` parameter from the request would be different, but you'd be using an old piece). The only way this has performance gains is if you call `set_piece` multiple times in the same request; and if you do this, you shouldn't.

Do you call `set_piece` multiple times in the same request?